Spigot-Server-Patches/0349-Use-ConcurrentHashMap-in-JsonList.patch

```diff
@@ -83,38 +84,46 @@ index 0859c7eb..e92b4c7b 100644
 +        return this.getBackingMap().isEmpty(); // Paper - readability is the goal. As an aside, isEmpty() uses only sumCount() and a comparison. size() uses sumCount(), casts, and boolean logic
      }
  
++    protected final String getMappingKey(K k0) { return a(k0); } // Paper - OBFHELPER
      protected String a(K k0) {
          return k0.toString();
      }
-+    protected final String getMappingKey(K k0) { return k0.toString(); } // Paper - OBFHELPER
- 
-     protected boolean d(K k0) {
+@@ -120,8 +128,10 @@ public class JsonList<K, V extends JsonListEntry<K>> {
          return this.d.containsKey(this.a(k0));
      }
  
++    private void removeStaleEntries() { h(); } // Paper - OBFHELPER
      private void h() {
 -        ArrayList arraylist = Lists.newArrayList();
 +        // Paper start
 +        /*ArrayList arraylist = Lists.newArrayList();
```

Comment:
```
/* Paper start
ArrayList...
```
Might be better idea?

Reply:
this is fine, think we even recommend this in the contrib guide

Reply:
Yep, nabbed the formatting from the guide and a few examples
https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md#formatting

Reply:
I'd personally aim for minimal diff (only additions where possible), but okay :D